src/BuildScriptGenerator/DotNetCore/DotnetCorePlatform.cs

```diff
@@ -80,19 +80,27 @@ public IEnumerable<string> SupportedVersions
         /// <inheritdoc/>
         public PlatformDetectorResult Detect(RepositoryContext context)
         {
-            var detectionResult = _detector.Detect(new DetectorContext
+            try
             {
-                SourceRepo = new Detector.LocalSourceRepo(context.SourceRepo.RootPath),
-            });
+                var detectionResult = _detector.Detect(new DetectorContext
+                {
+                    SourceRepo = new Detector.LocalSourceRepo(context.SourceRepo.RootPath),
+                });
+                if (detectionResult == null)
+                {
+                    return null;
+                }
 
-            if (detectionResult == null)
+                var version = ResolveVersion(detectionResult.PlatformVersion);
+                detectionResult.PlatformVersion = version;
+                return detectionResult;
+            }
+            catch (InvalidProjectFileException e)
             {
-                return null;
+                _logger.LogError(e, "InvalidProjectFileException caught: Either could not find the .NET Core project file" +
+                    "or found ambiguity in selecting a project to build.");
+                throw new InvalidUsageException($"Invalid project file exception caught from IDotNetCorePlatformDetector. ");
```

Comment:
Note that the message that we put in `InvalidUsageException` should be something which can be acted upon by an end user...You can just assign the message of the `InvalidProjectFileException` here.